addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -578,18 +578,34 @@ export class PosStore extends Reactive {
 
             if (payload) {
                 // Find candidate based on instantly created variants.
-                const instantlySelectedIds = this.models["product.template.attribute.value"]
+                const attributeValues = this.models["product.template.attribute.value"]
                     .readMany(payload.attribute_value_ids)
-                    .filter((value) => value.attribute_id.create_variant === "always")
                     .map((value) => value.id);
 
-                const candidate = productTemplate.product_variant_ids.find((variant) => {
+                let candidate = productTemplate.product_variant_ids.find((variant) => {
                     const attributeIds = variant.product_template_variant_value_ids.map(
                         (value) => value.id
                     );
-                    return instantlySelectedIds.every((id) => attributeIds.includes(id));
+                    return (
+                        attributeValues.every((id) => attributeIds.includes(id)) &&
+                        attributeValues.length
+                    );
                 });
 
+                const isDynamic = productTemplate.attribute_line_ids.some(
+                    (line) => line.attribute_id.create_variant === "dynamic"
+                );
```

Comment:
You can put this inside the if under so that is it not computed if candidate is not undefined